src/effects/Distortion.cpp

```diff
@@ -277,17 +377,29 @@ OptionalMessage EffectDistortion::DoLoadFactoryPreset(int id)
    }
 
    mSettings = FactoryPresets[id].params;
-   mThreshold = DB_TO_LINEAR(mSettings.mThreshold_dB);
+   mMaster.mThreshold = DB_TO_LINEAR(mSettings.mThreshold_dB);
 
    return { nullptr };
 }
 
 
 // Effect implementation
 
-std::unique_ptr<EffectUIValidator> EffectDistortion::PopulateOrExchange(
-   ShuttleGui & S, EffectInstance &, EffectSettingsAccess &,
-   const EffectOutputs *)
+std::unique_ptr<EffectUIValidator>
+EffectDistortion::PopulateOrExchange(ShuttleGui& S, EffectInstance& instance,
+   EffectSettingsAccess& access, const EffectOutputs* pOutputs)
+{
+   // TODO
+//   auto& settings = access.Get();
+//   auto& myEffSettings = GetSettings(settings);
+   auto& myEffSettings = mSettings;
+   auto result = std::make_unique<Validator>(*this, access, myEffSettings);
+   result->PopulateOrExchange(S);
+   return result;
+}
+
+// STEP TODO: add all links to OnSomething handlers
```

Comment:
I don't understand the comment

Reply:
It was deleted later, just a reminder for me to do something and I forgot it there.